flink-end-to-end-tests/flink-batch-sql-test/src/test/java/org/apache/flink/sql/tests/Generator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.sql.tests;
+
+import org.apache.flink.table.data.RowData;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+class Generator implements Iterator<RowData>, Iterable<RowData> {
+    final int numKeys;
+
+    private int keyIndex = 0;
+
+    private final long durationMs;
+    private final long stepMs;
+    private final long offsetMs;
+    private long ms = 0;
+
+    static Generator create(
+            int numKeys, float rowsPerKeyAndSecond, int durationSeconds, int offsetSeconds) {
+        int sleepMs = (int) (1000 / rowsPerKeyAndSecond);
+        return new Generator(numKeys, durationSeconds * 1000L, sleepMs, offsetSeconds * 2000L);
+    }
+
+    Generator(int numKeys, long durationMs, long stepMs, long offsetMs) {
+        this.numKeys = numKeys;
+        this.durationMs = durationMs;
+        this.stepMs = stepMs;
+        this.offsetMs = offsetMs;
+    }
+
+    @Override
+    public boolean hasNext() {
+        return ms < durationMs;
+    }
+
+    @Override
+    public RowData next() {
+        if (!hasNext()) {
+            throw new NoSuchElementException();
+        }
+        RowData row =
+                new GeneratedRow(
+                        keyIndex,
+                        LocalDateTime.ofInstant(
+                                Instant.ofEpochMilli(ms + offsetMs), ZoneOffset.UTC),
+                        "Some payload...");
```

Comment:
I saw this was in the original change, but should we randomize the payload?

Reply:
The goal of this PR is to port the test.

I preserved its logic 100%, so, I left the record generation as it was in the original code 👍